### DISCUSSION TITLE
Gitlab Projects based on templates

### DISCUSSION THREAD

**Discussion by nightHawk13998** [python-gitlab/python-gitlab#2267] [Q&A] [answered]
Hi, I am new to using this python wrapper and can't seem to figure out to create a project and have it use a custom template.  I know the API supports using '--use-custom-template' but cant seem to figure out the correct call using this wrapper.  If anyone has any suggestions that would be great. 

`
group_id = XXXXXX
template = "my_custom_template"
project = gl.projects.create({'name': 'test_repo', 'namespace_id': group_id, 'use_custom_template"': template})
`

**Comment by raedma** [comment, marked as answer]
I faced the same problem. The [Gitlab API description](https://python-gitlab.readthedocs.io/en/stable/cli-objects.html#gitlab-project-create) says there is the option `--use_custom_template` as well as `--group-with-project-templates-id` and `--template-project-id` so I tried my luck with

    project = gitlab.projects.create({'name': 'name', 'namespace_id': subgroup.id, 'use_custom_template': True, 'group_with_project_templates_id': 42540,'template_project_id': 30982})

and this worked fine.
